django/chat/responses.py

```diff
@@ -141,54 +134,76 @@ def summarize_response(chat, response_message):
     target_language = chat.options.summarize_language
     model = chat.options.summarize_model
 
+    llm = OttoLLM(model)
+
+    # TODO: Extracting text from file may incur Azure Document AI costs.
+    # Need to refactor extract_text to create Cost object with correct user and mode.
     async def multi_summary_generator():
+        full_text = ""
         for i, file in enumerate(files):
-            yield f"**{file.filename}**\n\n"
+            full_text += f"**{file.filename}**\n\n"
+            yield full_text
             if not file.text:
                 await sync_to_async(file.extract_text)(fast=True)
-            summary = await summarize_long_text_async(
+            response_stream = await summarize_long_text_async(
```

Comment:
Streaming summaries now, even for multiple file uploads.

This can be parallelized but I didn't want to get into it right now. The idea would be combining multiple streaming responses (for different documents or chunks). I have discussed this with Shaun to implement in Advanced Q&A since it is highly applicable there.